core/job/service/job_service.go

```diff
@@ -178,13 +178,19 @@ func (j *JobService) Add(ctx context.Context, jobTenant tenant.Tenant, specs []*
 	addedJobs, err := j.jobRepo.Add(ctx, jobs)
 	me.Append(err)
 
-	jobsWithUpstreams, err := j.upstreamResolver.BulkResolve(ctx, jobTenant.ProjectName(), addedJobs, logWriter)
+	downstreamJobs, err := j.getDownstreamJobs(ctx, jobTenant.ProjectName(), jobs)
+	me.Append(err)
+
+	jobsToBeResolved := []*job.Job{}
+	jobsToBeResolved = append(jobsToBeResolved, addedJobs...)
+	jobsToBeResolved = append(jobsToBeResolved, downstreamJobs...)
+	jobsWithUpstreams, err := j.upstreamResolver.BulkResolve(ctx, jobTenant.ProjectName(), jobsToBeResolved, logWriter)
```

Comment:
Nit: Can be shortened as `jobsWithUpstreams, err := j.upstreamResolver.BulkResolve(ctx, jobTenant.ProjectName(), append(addedJobs, downstreamJobs...), logWriter)